lib/logic/box_communicator.dart

```diff
@@ -81,6 +87,110 @@ class BoxCommunicator {
     box.close();
   }
 
+  //This function is called when a button on the settings page is pressed. After that
+  //the method will be executed. Normally all files and chunks will be downloaded that
+  //that arent on device yet.
+  void downloadAllData(BuildContext context) async {
+    _executeDownload = false;
+    print("Downloading All Data ...");
+    context.read<DownloadAllState>().downloading();
+    if (storage == null) storage = await Hive.openBox('storage');
+
+    String boxName;
+    int deviceTimestamp;
+    List idList = List();
+    Map<String, String> headers = {
+      "Content-type": "application/json",
+    };
+
+    //register to get the current boxName and get a id(timestamp) from the Sensorbox
+    final boxResponse = await retry(
+        () => http.get(boxIP + "/api/register").timeout(Duration(seconds: 3)),
+        retryIf: (e) => e is SocketException || e is TimeoutException);
+    if (boxResponse.statusCode == 200) {
+      var res = jsonDecode(boxResponse.body);
+      boxName = res["piID"];
+      deviceTimestamp = res["timestamp"];
+      //storage box stores the number of received bytes at 'totalSizeInBytes'
+      //opens the box for the current connected Sensorbox
+      if (box == null) {
+        box = await Hive.openLazyBox(boxName);
+      }
+      print("Opened box with boxname " + boxName.toString());
+      //If List of IDs is empty execute the download without a data limit and without priority restrictions
+      if (box.keys.toList().length > 0) {
```

Comment:
just use box.length in this case

Reply:
I think it doesn't matter.

Reply:
if it doesn't matter then you should use box.length since its shorter and neater